server/app/durablejobs/jobs/FixApplicantDobDataPathJob.java

```diff
@@ -0,0 +1,51 @@
+package durablejobs.jobs;
+
+import com.google.common.base.Preconditions;
+import durablejobs.DurableJob;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.time.LocalDate;
+import java.util.List;
+import models.Applicant;
+import models.PersistedDurableJob;
+import services.WellKnownPaths;
+import services.applicant.ApplicantData;
+
+/**
+ * Finds applicant data objects with applicant_date_of_birth set to a number and changes it to a
+ * nest object.
+ */
+public final class FixApplicantDobDataPathJob extends DurableJob {
+  private final Database database;
+
+  private final PersistedDurableJob persistedDurableJob;
+
+  public FixApplicantDobDataPathJob(PersistedDurableJob persistedDurableJob) {
+    this.persistedDurableJob = Preconditions.checkNotNull(persistedDurableJob);
+    this.database = DB.getDefault();
+  }
+
+  @Override
+  public PersistedDurableJob getPersistedDurableJob() {
+    return persistedDurableJob;
+  }
+
+  @Override
+  public void run() {
+    List<Applicant> applicants = database.find(Applicant.class).findList();
```

Comment:
suggestion (blocking): Loading all of the applicants into memory may be enough to cause an outage. Use a streaming query for O(1) memory consumption: https://ebean.io/docs/query/streaming

Reply:
postgres=# select  ((object #>> '{}')::jsonb)::json#>'{applicant,applicant_date_of_birth}' from applications;
                                   ?column?                                    
-------------------------------------------------------------------------------
 
 
 
 {"date": 1693094400000, "updated_at": 1699467429935, "program_updated_in": 4}
 {"date": 1690675200000, "updated_at": 1699467457146, "program_updated_in": 4}
(5 rows)

Reply:
Try something like below Rocky - select  ((object #>> '{}')::jsonb)::json#>'{applicant,applicant_date_of_birth,date}' from applications;

Reply:
Thank you both so much! I re-wrote this using your suggestions and am feeling much better about it.